client/express-checkout/utils/checkPaymentMethodIsAvailable.js

```diff
@@ -14,71 +14,75 @@ import WCPayAPI from 'wcpay/checkout/api';
 import { getUPEConfig } from 'wcpay/utils/checkout';
 
 export const checkPaymentMethodIsAvailable = memoize(
-	( paymentMethod, cart, resolve ) => {
-		// Create the DIV container on the fly
-		const containerEl = document.createElement( 'div' );
+	( paymentMethod, cart ) => {
+		return new Promise( ( resolve ) => {
```

Comment:
This is a good point. I was looking at it but haven't included it into this PR. I think that caching itself is a pretty nice addition because otherwise, like you're mentioning, we will be making unnecessary third-party Stripe requests.

There is indeed a potential issue because at the moment, `memoize` uses the entire `cart` object for cache key generation, which doesn't work well for object comparison and could lead to improper cache hits/misses.

By making the cache key more specific with only the relevant fields that affect the payment method availability:

```js
export const checkPaymentMethodIsAvailable = memoize(
	( paymentMethod, cart ) => {...},
	( paymentMethod, cart ) => {
		return JSON.stringify({
			paymentMethod,
			total_price: cart.cartTotals.total_price,
			country: cart.billingAddress?.country
		});
	}
);
```

we will ensure it runs on some specific updates. For that, we'd need to know what fields are relevant in our context, e.g. if billing country change should trigger reviewing ece availability.